File: android/src/main/java/org/maplibre/reactnative/components/styles/sources/MLRNRasterSource.java
package org.maplibre.reactnative.components.styles.sources;

import android.content.Context;

import org.maplibre.geojson.Feature;
import org.maplibre.android.style.sources.RasterSource;

import static org.maplibre.android.style.sources.RasterSource.DEFAULT_TILE_SIZE;

public class MLRNRasterSource extends MLRNTileSource<RasterSource> {
    private Integer mTileSize;

    public MLRNRasterSource(Context context) {
        super(context);
    }

    @Override
    public RasterSource makeSource() {
        String configurationUrl = getURL();
        int tileSize = mTileSize == null ? DEFAULT_TILE_SIZE : mTileSize;
        if (configurationUrl != null) {
            return new RasterSource(mID, configurationUrl, tileSize);
        }
        return new RasterSource(mID, buildTileset(), tileSize);
    }


    public void setTileSize(int tileSize) {
        mTileSize = tileSize;
    }

    @Override
    public boolean hasPressListener() {
        return false;
    }

    @Override
    public void onPress(OnPressEvent feature) {
        // ignore, cannot query raster layers
    }
}

File: android/src/main/java/org/maplibre/reactnative/components/sources/tilesources/rastersource/MLRNRasterSource.kt
package org.maplibre.reactnative.components.sources.tilesources.rastersource

import android.content.Context
import org.maplibre.android.style.sources.RasterSource
import org.maplibre.reactnative.components.sources.tilesources.MLRNTileSource

class MLRNRasterSource(context: Context?) : MLRNTileSource<RasterSource?>(context) {
    var tileSize: Int? = null

    override fun makeSource(): RasterSource {
        val configurationUrl = url
        val tileSize = (if (tileSize != null) tileSize else RasterSource.DEFAULT_TILE_SIZE)!!
        if (configurationUrl != null) {
            return RasterSource(mID, configurationUrl, tileSize)
        }
        return RasterSource(mID, buildTileset(), tileSize)
    }
}
